app/controllers/FormController.scala

```diff
@@ -7,24 +7,25 @@ import play.api.data.Forms._
 import play.api.mvc._
 import play.api.data.validation.Constraints._
 import scala.collection.mutable.ListBuffer
-import java.time.LocalDateTime
-import java.time.format.DateTimeFormatter
+import java.time.LocalDate
+import java.time.LocalTime
+import java.time.temporal.ChronoUnit
 
-case class DataModel(name: String, age: String, message: String, timestamp: Option[String], datestamp: Option[String])
+case class DataModel(name: String, age: String, message: String, timestamp: Option[LocalTime], datestamp: Option[LocalDate])
```

Comment:
could consider replacing timestamp and datestamp with a single LocalDateTime (or Instant). you can always separate out the date and time later on using different formatters

Reply:
Thanks for the pointer, I've refactored my code further to do this. I set a single LocalDateTime field in my database, and then have a seperate helper object which I call from my View template to format the data.